multi/imxrt-multi/config.h

```diff
@@ -26,186 +28,668 @@
 #include "imxrt1060.h"
 #endif
 
+#include <board_config.h>
+
+
 #ifndef IMXRT_MULTI_PRIO
 #define IMXRT_MULTI_PRIO 2
 #endif
 
 /* UART */
 
 #ifndef UART1
-#define UART1 1
+#define UART1          0
+#define UART1_BAUDRATE 0
+#elif !ISBOOLEAN(UART1)
+#error "UART1 must have a value of 0, 1, or be undefined"
+#endif
+
+#if UART1
+
+#ifndef UART1_TX_PIN
+#warning "[Deprecated] UART1_TX_PIN should be explicitly defined in board_config.h. Using the default"
+#define UART1_TX_PIN UART1_TX_PIN_DEFAULT
+#elif ISEMPTY(UART1_TX_PIN)
+#error "UART1_TX_PIN must not be empty"
+#endif
+
+#ifndef UART1_RX_PIN
+#warning "[Deprecated] UART1_RX_PIN should be explicitly defined in board_config.h. Using the default"
+#define UART1_RX_PIN UART1_RX_PIN_DEFAULT
+#elif ISEMPTY(UART1_RX_PIN)
+#error "UART1_RX_PIN must not be empty"
 #endif
 
-#ifndef UART1_HW_FLOWCTRL
-#define UART1_HW_FLOWCTRL 0
+#ifdef UART1_HW_FLOWCTRL
+#warning "[Deprecated] UART1_HW_FLOWCTRL should not be used. Defines for UART1_RTS_PIN and/or UART1_CTS_PIN are enough"
+#if ISTRUTHY(UART1_HW_FLOWCTRL)
+#define UART1_RTS_PIN UART1_RTS_PIN_DEFAULT
+#define UART1_CTS_PIN UART1_CTS_PIN_DEFAULT
+#else
+#define UART1_RTS_PIN
+#define UART1_CTS_PIN
+#endif
+#endif
+
+#ifndef UART1_RTS_PIN
+#warning "[Deprecated] UART1_RTS_PIN should be explicitly defined (may be empty) in board_config.h. Using the default"
+#define UART1_RTS_PIN UART1_RTS_PIN_DEFAULT
+#endif
+
+#ifndef UART1_CTS_PIN
+#warning "[Deprecated] UART1_CTS_PIN should be explicitly defined (may be empty) in board_config.h. Using the default"
+#define UART1_CTS_PIN UART1_CTS_PIN_DEFAULT
 #endif
 
 #ifndef UART1_BAUDRATE
+#warning "[Deprecated] UART1_BAUDRATE should be explicitly defined in board_config.h. Using the default"
 #define UART1_BAUDRATE 115200
 #endif
 
+#endif
```

Comment:
as we're not using indentation for ifdefs (maybe worth reconsidering?) - write what context we're exiting -  `/* UART1 */`

Reply:
Done